genesis/ext/pyrender/viewer.py

```diff
@@ -556,6 +557,8 @@ def save_video(self, filename=None):
         """
         if filename is None:
             filename = self._get_save_filename(["mp4"])
```

Comment:
Why this is not enough to get a valid filename? Anyway, we should at least raise an exception if fieldname is None because the user closed the file dialog.